src/Symfony/Component/HttpKernel/Fragment/AbstractSurrogateFragmentRenderer.php

```diff
@@ -64,6 +64,10 @@ public function __construct(SurrogateInterface $surrogate = null, FragmentRender
     public function render($uri, Request $request, array $options = array())
     {
         if (!$this->surrogate || !$this->surrogate->hasSurrogateCapability($request)) {
+            if ($uri instanceof ControllerReference && $this->containsNonScalars($uri->attributes)) {
+                @trigger_error('Passing objects as part of URI attributes to the ESI and SSI rendering strategies is deprecated. Use another rendering strategy or use scalar values.', E_USER_DEPRECATED);
```

Comment:
you should mention that this is deprecated as of 3.1 and that this possibility will be removed in 4.0.